### PR TITLE
[PROD-793] Add collapsible detail info for pool card

### PR DESCRIPTION
**Pull request by leqiang-deltafi**
The detailed info can collapse/expand when clicking the 'detail'
This part is hidden under mainnet-prod, we need to polish to style and layout more
Screen Shot 1401-04-02 at 23 19 37
Screen Shot 1401-04-02 at 23 19 49

magnifier = 1 + (virtualReservePercentage/100)

### CODE REVIEW

File: src/views/Pool/utils.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { SwapInfo } from "anchor/type_definitions";
+import BigNumber from "bignumber.js";
+import { TokenConfig } from "constants/deployConfigV2";
+import { getTokenTvl } from "utils/utils";
+
+export function calculatePoolTvl(
```
Comment (leqiang-deltafi) [inline src/views/Pool/utils.ts:6]: this logic is used in both pool and dashboard, put it to a util function of pool page

**Comment by ypeng-deltafi** [comment]
@leqiang-deltafi I think we can remove the account addresses. They are not very useful and other DEX doesn't show them.

**Comment by leqiang-deltafi** [comment]
> @leqiang-deltafi I think we can remove the account addresses. They are not very useful and other DEX doesn't show them.

ok, removed

File: src/components/PoolStats/PoolStats.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        {isExpanded ? "Details" : "Details ..."}
+      </Box>
+      <Box {...getCollapseProps()}>
+        <Box className="content">
+          <Box className={classes.dataRow}>
+            <Box textAlign={"left"}>Currency Reserves:</Box>
+            <Box textAlign={"right"}>
+              <Box display={"flex"} justifyContent={"space-between"}>
+                <Box>
+                  {currencyValueDisplay(baseReserve.toFixed(0))} {baseTokenInfo.symbol}
```
Comment (ypeng-deltafi) [inline src/components/PoolStats/PoolStats.tsx:68]: toFixed(0) will give 0 on most of our pool. can you add more digits?

File: src/components/PoolStats/PoolStats.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                  sx={{
+                    height: 20,
+                    width: 20,
+                  }}
+                  src={baseTokenInfo.logoURI}
+                />
+              </Box>
+              <Box display={"flex"}>
+                <Box>
+                  {currencyValueDisplay(quoteReserve.toFixed(0))} {quoteTokenInfo.symbol}{" "}
```
Comment (ypeng-deltafi) [inline src/components/PoolStats/PoolStats.tsx:80]: ditto.

File: src/components/PoolStats/PoolStats.tsx (hunk 2 above)
Comment (leqiang-deltafi) [reply src/components/PoolStats/PoolStats.tsx:68]: I think we can just display the integer part for simplicity just like saber. The reserve is supposed to be a large number

Comment (ypeng-deltafi) [reply src/components/PoolStats/PoolStats.tsx:68]: The problem is that our initial deposit to the prod is also less than 1. If we want do this, we should change our deployment to create initial deposit larger than 1

Comment (ypeng-deltafi) [reply src/components/PoolStats/PoolStats.tsx:68]: from my perspective, it is very wierd to see a pool with `0` reserve

Comment (ypeng-deltafi) [reply src/components/PoolStats/PoolStats.tsx:68]: it is very confusing

Comment (leqiang-deltafi) [reply src/components/PoolStats/PoolStats.tsx]: added a conditional display, if either value is less than 100, we display the full decimals, if both are greater than 100, we display only integer part

Comment (ypeng-deltafi) [reply src/components/PoolStats/PoolStats.tsx]: sg

**Comment by ypeng-deltafi** [comment]
@leqiang-deltafi I think the display for the floating part is not corrot. We shouldn't add `,` there. 
67DDDB99-B6DF-4A42-81CB-EF52CE994F98_4_5005_c

**Comment by leqiang-deltafi** [comment]
> @leqiang-deltafi I think the display for the floating part is not corrot. We shouldn't add `,` there. 67DDDB99-B6DF-4A42-81CB-EF52CE994F98_4_5005_c

I see, updated the value formatting method

File: src/utils/utils.ts [hunk 4]
```diff
@@ -5,12 +5,22 @@ import { TokenConfig } from "constants/deployConfigV2";
 import { anchorBnToBn } from "./tokenUtils";
 import { SwapInfo } from "anchor/type_definitions";
 
+export const currencyValueDisplay = (value) => {
+  const valueParts = value.toString().split(".");
+  if (!valueParts[0]) {
+    return null;
+  }
+  valueParts[0] = valueParts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
Comment (leqiang-deltafi) [inline src/utils/utils.ts:13]: only add separator to the integer part

File: src/utils/utils.ts [hunk 5]
```diff
@@ -5,8 +5,15 @@ import { TokenConfig } from "constants/deployConfigV2";
...
 
-export const currencyValueDisplay = (value) =>
-  value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+export const currencyValueDisplay = (value) => {
+  const valueParts = value.toString().split(".");
+  if (!valueParts[0]) {
+    return null;
+  }
+  valueParts[0] = valueParts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+  console.log(valueParts);
```
Comment (ypeng-deltafi) [inline src/utils/utils.ts:14]: remove the log

**Review by ypeng-deltafi — APPROVED** [review]
please also fix the linter

**Comment by leqiang-deltafi** [comment]
> please also fix the linter

updated

**Pull request merged**